capa/ida/plugin/view.py

```diff
@@ -210,12 +213,99 @@ def load_preview_meta(self, ea, author, scope):
         self.setText("\n".join(metadata_default))
 
     def keyPressEvent(self, e):
-        """ """
-        if e.key() == QtCore.Qt.Key_Tab:
-            self.insertPlainText(" " * 2)
+        """intercept key press events"""
+        if e.key() in (QtCore.Qt.Key_Tab, QtCore.Qt.Key_Backtab):
+            # apparently it's not easy to implement tabs as spaces, or multi-line tab or SHIFT + Tab
+            # so we need to implement it ourselves so we can retain properly formatted capa rules
+            # when a user uses the Tab key
+            if self.textCursor().selection().isEmpty():
+                # single line, only worry about Tab
+                if e.key() == QtCore.Qt.Key_Tab:
+                    self.insertPlainText(self.INDENT)
```

Comment:
Is this needed? When nothing is selected tab already work 🤔

Reply:
It may be needed because this handles all `tab` key presses.